modules/linear_boltzmann_solvers/discrete_ordinates_problem/discrete_ordinates_problem.cc

```diff
@@ -80,31 +77,58 @@ DiscreteOrdinatesProblem::DiscreteOrdinatesProblem(const InputParameters& params
     verbose_sweep_angles_(params.GetParamVectorValue<int>("directions_sweep_order_to_print")),
     sweep_type_(params.GetParamValue<std::string>("sweep_type"))
 {
-  scattering_order_ = params.GetParamValue<int>("scattering_order");
-  ValidateAndComputeScatteringMoments();
-
   if (use_gpus_ && sweep_type_ == "CBC")
   {
     log.Log0Warning() << "Sweep computation on GPUs is not supported for the CBC sweep."
                       << "Falling back to CPU sweep.\n";
     use_gpus_ = false;
   }
 
+  // Chceck for consistency between quadrature sets
   for (auto& groupset : groupsets_)
   {
-    // Build groupset angular flux unknown manager
     if (not groupset.quadrature)
     {
       std::stringstream oss;
       oss << GetName() << ":\nGroupset " << groupset.id
           << " does not have an associated quadrature set";
       throw std::runtime_error(oss.str());
     }
+
+    if (groupset.quadrature->GetScatteringOrder() != groupset.quadrature->GetScatteringOrder())
```

Comment:
I do not get this. Are you querying the same thing twice? It will never be different.